src/modules/sway/bar.cpp

```diff
@@ -61,38 +105,124 @@ void BarIpcClient::onInitialConfig(const struct Ipc::ipc_response& res) {
+    switch (res.type) {
+      case IPC_EVENT_WORKSPACE:
+        if (payload.isMember("change")) {
+          // only check and send signal if the workspace update reason was because of a urgent
+          // change
+          if (payload["change"] == "urgent") {
+            auto urgent = payload["current"]["urgent"];
+            if (urgent.asBool()) {
+              // Event for a new urgency, update the visibly
+              signal_urgency_(true);
+            } else if (!urgent.asBool() && visible_by_urgency_) {
+              // Event clearing an urgency, bar is visible, check if another workspace still has
+              // the urgency hint set
+              ipc_.sendCmd(IPC_GET_WORKSPACES);
+            }
+          }
+          last_press_modifier_ = false;
+        }
+        break;
+      case IPC_EVENT_MODE:
+        if (payload.isMember("change")) {
+          signal_mode_(payload["change"] != "default");
+          last_press_modifier_ = false;
+        }
+        break;
+      case IPC_EVENT_BINDING:
+        last_press_modifier_ = false;
+        break;
+      case IPC_EVENT_BAR_STATE_UPDATE:
+      case IPC_EVENT_BARCONFIG_UPDATE:
+        if (auto id = payload["id"]; id.isString() && id.asString() != bar_.bar_id) {
+          spdlog::trace("swaybar ipc: ignore event for {}", id.asString());
+          return;
+        }
+        if (payload.isMember("visible_by_modifier")) {
+          // visibility change for hidden bar
+          signal_visible_(payload["visible_by_modifier"].asBool());
+        } else {
+          // configuration update
+          auto config = parseConfig(payload);
+          signal_config_(std::move(config));
+        }
+        break;
     }
   } catch (const std::exception& e) {
     spdlog::error("BarIpcClient::onEvent {}", e.what());
   }
 }
 
+void BarIpcClient::onCmd(const struct Ipc::ipc_response& res) {
+  if (res.type == IPC_GET_WORKSPACES) {
+    try {
+      auto payload = parser_.parse(res.payload);
+      for (auto& ws : payload) {
+        if (ws["urgent"].asBool()) {
+          spdlog::debug("Found workspace {} with urgency set. Stopping search.", ws["name"]);
+          // Found one workspace with urgency set, signal bar visibility
+          signal_urgency_(true);
+          return;
+        }
+      }
+      // Command to get workspaces was sent after a change in workspaces was based on "urgent",
+      // if no workspace has this flag set to true, all flags must be cleared.
+      signal_urgency_(false);
+    } catch (const std::exception& e) {
+      spdlog::error("Bar: {}", e.what());
+    }
+  }
+}
+
 void BarIpcClient::onConfigUpdate(const swaybar_config& config) {
   spdlog::info("config update for {}: id {}, mode {}, hidden_state {}", bar_.bar_id, config.id,
                config.mode, config.hidden_state);
   bar_config_ = config;
+
+  // Get this option from the local configuration
+  bar_config_.modifier_reset = this->bar_.config.get("modifier-reset", "press").asString();
```

Comment:
This option cannot be modified by an IPC message. Swaybar IPC events are limited to what sway itself supports and cannot be expanded.

`bar_.config` is a waybar json config which doesn't support live reload, so you can read the option once in the constructor and move the property to the `BarIpcClient` class.

Reply:
Was in the class for my first implementation :) but moved it to `bar_config_` so it would all be in the same place. Anyway, back in the class, looks cleaner as well.